### ISSUE TITLE
Hitting the button to announce loot does nothing

### ISSUE THREAD

**Issue by rfevang** [Alumian/CEPGP-Retail#278]
**The Error**

No error, but nothing happened (there is a LUA error that doesn't show normally). LUA error:

```
Message: Interface\AddOns\CEPGP\Loot.lua:107: attempt to index field 'RaidVisibility' (a boolean value)
Time: Tue Aug  4 21:45:57 2020
Count: 6
Stack: Interface\AddOns\CEPGP\Loot.lua:107: attempt to index field 'RaidVisibility' (a boolean value)
Interface\AddOns\CEPGP\Loot.lua:107: in function `CEPGP_announce'
Interface\AddOns\CEPGP\Utility.lua:790: in function <Interface\AddOns\CEPGP\Utility.lua:784>

Locals: link = "[High Councillor's Circlet of Fiery Wrath]"
x = 1
slotNum = 1
quantity = 1
iString = "item:10139::::::1901:217259136:60:::::::[High Councillor's Circlet of Fiery Wrath]|"
name = "High Councillor's Circlet of Fiery Wrath"
_ = "[High Councillor's Circlet of Fiery Wrath]"
_ = 2
_ = 62
_ = 57
_ = "Armor"
_ = "Cloth"
_ = 1
slot = "INVTYPE_HEAD"
tex = 132768
id = "10139"
temp = <table> {
}
schema = "lootschema;1;Main Spec;2;Off Spec;3;Roll List;4;Minor Upgrade;5;;6;Pass"
temp = <table> {
 1 = "lootschema;1;Main Spec;2;Off Spec;3;Roll List;4;Minor Upgrade;5;;6;Pass"
}
(for generator) = <function> defined =[C]:-1
(for state) = <table> {
 1 = "lootschema;1;Main Spec;2;Off Spec;3;Roll List;4;Minor Upgrade;5;;6;Pass"
}
(for control) = 1
_ = 1
schema = "lootschema;1;Main Spec;2;Off Spec;3;Roll List;4;Minor Upgrade;5;;6;Pass"
(*temporary) = true
(*temporary) = 1
(*temporary) = "lootschema;1;Main Spec;2;Off Spec;3;Roll List;4;Minor Upgrade;5;;6;Pass"
(*temporary) = ";"
(*temporary) = "Pass"
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index field 'RaidVisibility' (a boolean value)"
```

**Describe the bug**
After updating to 1.12.22 (I'm not sure what the previous version was), hitting the button to announce loot stopped working. It still worked for a different master looter in the guild, though he has some settings different (not sure exactly which)

**To Reproduce**
Not sure what triggers it, but it does not work for me at all at the moment

**Debug Information**

<details><summary>Debug Info</summary><br />
Version: 1.12.22 Release<br />
Locale: enUS<br />
Poll Rate: 0.0001<br />
GP Formula: (4.83x(2^<sup>((ilvl/26)+(rarity-4))</sup>)xSlot Modifier)x1<br />
Base GP: 10<br />
Traffic Entries: 2259<br />
Base GP Decay Factor: false<br />
Standby EP: False<br />
Standby Offline: False<br />
Standby Percent: 100<br />
Standby Accept Whispers: False<br />
Standby EP by Rank: False<br />
Standby EP Manual Delegation: True<br />
Standby EP Whisper Keyphrase: standby<br />
Auto Pass on Ineligible Items: false<br />
Full Raid Loot Visibility: true<br />
GP on Tooltips: true<br />
Suppress Loot Announcements: false<br />
Roll for Passes: false<br />
Announce Rolls: true<br />
Resolve Duplicate Rolls: true<br />
Minimum EP: false, 0<br />
Reporting Channel: Guild<br />
Loot Response Channel: Officer<br />
Minimum Threshold for Loot: 4<br />
Allow Force Sync: false, GM<br />
Hide Keyphrase Alternatives: False<br /><br /><details><summary>Auto EP</summary>
High Priestess Jeklik: False<br />
Gothik the Harvester: True<br />
Buru the Gorger: True<br />
Wushoolay: False<br />
Silithid Royalty: True<br />
Ossirian the Unscarred: True<br />
Golemagg the Incinerator: True<br />
Grand Widow Faerlina: True<br />
High Priestess Mar'li: False<br />
Twin Emperors: True<br />
Majordomo Executus: True<br />
High Priest Venoxis: False<br />
Gehennas: True<br />
Taerar: True<br />
Ysondre: True<br />
Thane Korth'azz: True<br />
High Priest Thekal: False<br />
Noth the Plaguebringer: True<br />
Lethon: True<br />
Moam: True<br />
The Prophet Skeram: True<br />
Flamegor: True<br />
C'Thun: True<br />
General Rajaxx: True<br />
The Edge of Madness: True<br />
Princess Huhuran: True<br />
Thaddius: True<br />
Loatheb: True<br />
The Silithid Royalty: True<br />
Ayamiss the Hunter: True<br />
Ebonroc: True<br />
Gahz'ranka: False<br />
Heigan the Unclean: True<br />
Azuregos: True<br />
Doom Lord Kazzak: True<br />
High Priestess Arlokk: False<br />
Baron Geddon: True<br />
Viscidus: True<br />
Gri'lek: False<br />
The Twin Emperors: True<br />
Jin'do the Hexxer: False<br />
Vaelastrasz the Corrupt: True<br />
Gluth: True<br />
Bloodlord Mandokir: False<br />
Firemaw: True<br />
Emeriss: True<br />
Shazzrah: True<br />
Sapphiron: True<br />
Battleguard Sartura: True<br />
Onyxia: False<br />
Kel'Thuzad: True<br />
Nefarian: True<br />
Fankriss the Unyielding: True<br />
Lucifron: True<br />
Anub'Rekhan: True<br />
Hazza'rah: False<br />
Ragnaros: True<br />
Ouro: True<br />
Broodlord Lashlayer: True<br />
Hakkar: False<br />
Maexxna: True<br />
The Four Horsemen: True<br />
Kurinnaxx: True<br />
Grobbulus: True<br />
Instructor Razuvious: True<br />
Razorgore the Untamed: True<br />
Magmadar: True<br />
Patchwerk: True<br />
Sulfuron Harbinger: True<br />
Renataki: False<br />
Chromaggus: True<br />
Garr: True<br />
</details>
<details><summary>EP Values</summary>
High Priestess Jeklik: 0<br />
Gothik the Harvester: 12<br />
Buru the Gorger: 3<br />
Wushoolay: 0<br />
Silithid Royalty: 10<br />
Ossirian the Unscarred: 4<br />
Golemagg the Incinerator: 5<br />
Grand Widow Faerlina: 12<br />
High Priestess Mar'li: 0<br />
Twin Emperors: 10<br />
Majordomo Executus: 5<br />
High Priest Venoxis: 0<br />
Gehennas: 5<br />
Taerar: 7<br />
Ysondre: 7<br />
Thane Korth'azz: 15<br />
High Priest Thekal: 0<br />
Noth the Plaguebringer: 12<br />
Lethon: 7<br />
Moam: 3<br />
The Prophet Skeram: 10<br />
Flamegor: 7<br />
C'Thun: 12<br />
General Rajaxx: 3<br />
The Edge of Madness: 2<br />
Princess Huhuran: 10<br />
Thaddius: 15<br />
Loatheb: 15<br />
The Silithid Royalty: 10<br />
Ayamiss the Hunter: 3<br />
Ebonroc: 7<br />
Gahz'ranka: 0<br />
Heigan the Unclean: 12<br />
Azuregos: 7<br />
Doom Lord Kazzak: 7<br />
High Priestess Arlokk: 0<br />
Baron Geddon: 5<br />
Viscidus: 10<br />
Gri'lek: 0<br />
The Twin Emperors: 10<br />
Jin'do the Hexxer: 0<br />
Vaelastrasz the Corrupt: 7<br />
Gluth: 12<br />
Bloodlord Mandokir: 0<br />
Firemaw: 7<br />
Emeriss: 7<br />
Shazzrah: 5<br />
Sapphiron: 15<br />
Battleguard Sartura: 10<br />
Onyxia: 0<br />
Kel'Thuzad: 15<br />
Nefarian: 10<br />
Fankriss the Unyielding: 10<br />
Lucifron: 5<br />
Anub'Rekhan: 12<br />
Hazza'rah: 0<br />
Ragnaros: 7<br />
Ouro: 10<br />
Broodlord Lashlayer: 7<br />
Hakkar: 0<br />
Maexxna: 15<br />
The Four Horsemen: 15<br />
Kurinnaxx: 3<br />
Grobbulus: 12<br />
Instructor Razuvious: 12<br />
Razorgore the Untamed: 7<br />
Magmadar: 5<br />
Patchwerk: 12<br />
Sulfuron Harbinger: 5<br />
Renataki: 0<br />
Chromaggus: 7<br />
Garr: 5<br />
</details>
<details><summary>Loot Button Configuration</summary>
Loot Button 1<br />State: Enabled<br />Button Text: Main Spec<br />Discount: 0%<br />Keyphrase Alternative: Need<br /><br />Loot Button 2<br />State: Enabled<br />Button Text: Off Spec<br />Discount: 75%<br />Keyphrase Alternative: Off<br /><br />Loot Button 3<br />State: Disabled<br />Button Text: Roll List<br />Discount: 100%<br />Keyphrase Alternative: Roll<br /><br />Loot Button 4<br />State: Disabled<br />Button Text: Minor Upgrade<br />Discount: 0%<br />Keyphrase Alternative: Minor<br /><br /></details><details><summary>Standby Guild Ranks</summary>
GM: False<br />
Officer: False<br />
Officer alt: False<br />
Raider: False<br />
Member: False<br />
Alt: False<br />
PVP: False<br />
Trial: False<br />
</details>
<details><summary>Slot Weights</summary>
2HWEAPON: 2<br />
NECK: 0.75<br />
HEAD: 1<br />
WEAPON: 1.5<br />
WRIST: 0.5<br />
WEAPONMAINHAND: 1.5<br />
SHIELD: 0.5<br />
HOLDABLE: 0.5<br />
CLOAK: 0.5<br />
THROWN: 0.5<br />
FEET: 0.75<br />
RELIC: 0.5<br />
FINGER: 0.75<br />
WEAPONOFFHAND: 1.5<br />
TRINKET: 1<br />
WAND: 0.5<br />
CHEST: 1<br />
ROBE: 1<br />
HAND: 0.75<br />
LEGS: 1<br />
WAIST: 0.75<br />
EXCEPTION: 1<br />
RANGED: 0.5<br />
SHOULDER: 0.75<br />
</details>
<details><summary>Override List</summary>
[Rejuvenating Gem]: 44<br />
[Spinal Reaper]: 40<br />
[Shadow Wing Focus Staff]: 40<br />
[Warblade of the Hakkari]: 14<br />
[Head of Onyxia]: 17<br />
[Claw of the Frost Wyrm]: 25<br />
[Arlokk's Grasp]: 13<br />
[The Eye of Divinity]: 32<br />
[Drake Talon Cleaver]: 40<br />
[Herald of Woe]: 40<br />
[Staff of Dominance]: 40<br />
[Ashjre'thul, Crossbow of Smiting]: 40<br />
[Drake Fang Talisman]: 44<br />
[Eskhandar's Left Claw]: 14<br />
</details>
<details><summary>Addons List</summary>
CEPGP<br />
ThreatClassic2<br />
WeakAuras<br />
DBM-Core<br />
DBM-DefaultSkin<br />
DBM-StatusBarTimers<br />
</details>
<details><summary>Enabled Plugins</summary>
</details>
</details>

**Comment by rfevang** [comment]
Might be related to the loot buttons somehow? Just opening the loot distribution tab of the options spawns an error with `/console scripterrors 1`: 

```
Message: [string "CEPGP_loot_options:OnShow"]:81: attempt to index field 'RaidVisibility' (a boolean value)
Time: Tue Aug  4 23:05:03 2020
Count: 4
Stack: [string "CEPGP_loot_options:OnShow"]:81: attempt to index field 'RaidVisibility' (a boolean value)
[string "*:OnShow"]:81: in function <[string "*:OnShow"]:1>
[C]: in function `Show'
Interface\FrameXML\InterfaceOptionsFrame.lua:204: in function `InterfaceOptionsFrame_Show'
[string "*:OnClick"]:1: in function <[string "*:OnClick"]:1>

Locals: self = CEPGP_loot_options {
 0 = <userdata>
 cancel = <function> defined @Interface\FrameXML\InterfaceOptionsFrame.lua:625
 name = "Loot Distribution"
 hidden = false
 parent = "Classic EPGP"
 refresh = <function> defined @Interface\FrameXML\InterfaceOptionsFrame.lua:627
 okay = <function> defined @Interface\FrameXML\InterfaceOptionsFrame.lua:624
 default = <function> defined @Interface\FrameXML\InterfaceOptionsFrame.lua:626
}
width = 550.000061
height = 254.425949
(*temporary) = <function> defined =[C]:-1
(*temporary) = CEPGP_loot_options_dist_assist_check {
 0 = <userdata>
}
(*temporary) = true
(*temporary) = <userdata>
(*temporary) = "CEPGP_loot_options"
(*temporary) = <function> defined =[C]:-1
(*temporary) = CEPGP_loot_options_4_whisper_edit {
 0 = <userdata>
 Right = <unnamed> {
 }
 Left = <unnamed> {
 }
 Middle = <unnamed> {
 }
}
(*temporary) = "Minor"
(*temporary) = <userdata>
(*temporary) = "_whisper_edit"
(*temporary) = "attempt to index field 'RaidVisibility' (a boolean value)"
```

**Comment by Alumian** [comment]
HI rfevang,

Thanks for submitting this bug report. This issue was caused by a data conversion issue. It has been fixed for the next release which will be soon, but for now, you can fix this issue by typing `/run CEPGP.Loot.RaidVisibility = {[1] = true, [2] = CEPGP_raid_wide_dist[2]}`.

Please let me know if that fixes your issue.

**Comment by Alumian** [comment]
Should be fixed in version 1.12.23. Please let me known if the issue persists.
